metals/src/main/scala/scala/meta/internal/metals/PackageProvider.scala

```diff
@@ -272,43 +276,117 @@ class PackageProvider(
+        }
+    val refsImportedViaWildCard =
+      refsNames -- directlyImportedNames
+        .withFilter(n => refsNames.contains(n.value))
+        .map(_.value)
+    lazy val directlyImportedSymbols =
+      directlyImportedNames
+        .flatMap(name =>
+          defProvider
+            .symbolOccurrence(source, name.pos.toLsp.getStart())
+        )
+        .map(_._1.symbol)
+        .toSet
+
+    val importEdits =
+      importersAndParts.map { case (importer, importParts) =>
+        calcImportEditsForFileMove(
+          importer,
+          importParts,
+          oldPkgParts,
+          refsNames,
+          refsSymbols,
+          newPkgName,
+          refsImportedViaWildCard,
+          () => directlyImportedSymbols,
+          source,
+        )
+      }
+    ImportEdits(
+      importEdits.flatMap(_.handledRefs).toList,
+      importEdits.flatMap(_.edits).toList,
+    )
+  }
+
   private def calcImportEditsForFileMove(
       importer: Importer,
       importParts: Vector[String],
       oldPkgParts: Vector[String],
       refsNames: Set[String],
+      refsSymbols: Set[String],
       newPkgName: String,
-  ): List[TextEdit] =
+      refsImportedViaWildCard: Set[String],
+      directlyImportedSymbols: () => Set[String],
+      source: AbsolutePath,
+  ): ImportEdits = {
+    def wildcardImportsOnlyRefs() = {
+      val importedSymbols = defProvider
+        .symbolOccurrence(source, importer.ref.pos.toLsp.getEnd())
+        .map { case (so, _) => so.symbol }
+        .toList
+        .flatMap(
+          refProvider
+            .referencesForWildcardImport(_, source, directlyImportedSymbols())
+        )
+      importedSymbols.forall(refsSymbols.contains(_))
+    }
     if (importParts == oldPkgParts) {
       val importees = importer.importees
       def isReferenced(importee: Importee) = importee match {
         case Importee.Name(name) => refsNames.contains(name.value)
         case Importee.Rename(from, _) => refsNames.contains(from.value)
         case Importee.Unimport(name) => refsNames.contains(name.value)
+        case Importee.Wildcard() if wildcardImportsOnlyRefs => true
         case _ => false
       }
       val (refsImportees, refsIndices) = importees.zipWithIndex.filter {
         case (i, _) =>
           isReferenced(i)
       }.unzip
       if (refsImportees.length == 0) {
-        List.empty
+        ImportEdits.empty
       } else {
-        val importeeStr = mkImportees(refsImportees.collect {
-          case Importee.Name(name) => name.value
+        def rep2(str: String) = (str, str)
+        val (handledRefs, importeeStrs) = refsImportees.flatMap {
+          case Importee.Name(name) => List(rep2(name.value))
           case Importee.Rename(from, to) =>
-            s"${from.value} => ${to.value}"
-          case Importee.Unimport(name) => name.value
-        })
+            List((from.value, s"${from.value} => ${to.value}"))
+          case Importee.Unimport(name) => List(rep2(name.value))
```

Comment:
Same here, we don't need to handle unimport I think? Potentially, this should actually remove from wilcard imports